web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1042,10 +1057,36 @@ namespace com.keyman.keyboards {
      **/
     register(x) {
       var options=x['options'];
+      // Represents the `resolve` function for the corresponding Promise built in `keymanCloudRequest`.
+      var sourcePromise: () => void;
+      const promiseid = x['timerid'];
+
+      let signalSuccess = function() {};
+      let signalFailure = function(err: string|Error) {};
```

Comment:
These signal functions feel awkward to me. Might it be clearer to wrap the function and test its result?

Something like:

```js
register(x) {
  const promiseid = x['timerid'];
  const promiseFuncs =  this.registrationResolvers[promiseid];
  const result = doRegister(x);
  if(promiseFuncs) {
    window.clearTimeout(promiseid);
    try {
      typeof result == string ? promiseFuncs.reject(err) : promiseFuncs.resolve();
    } finally {
      delete this.registrationResolvers[promiseid];
    }
  }
}

doRegister(x): string | boolean {
  ...
}
```